File: src/languagesMap/py/ListNode/implementation.py
class ListNode:
  def __init__(self, val=0, next=None):
    self.val = val
    self.next = next

def arrayToLinkedList(arrayNode):
  if len(arrayNode) == 0:
    return None
  head = ListNode()
  next = head
  length = len(arrayNode)
  for i in range(0, length):
    next.val = arrayNode[i]
    if i == length - 1:
      next.next = None
    else:
      next.next = ListNode()
      next = next.next
  return head

def linkedListToArray(head):
  array = []
  curr = head
  while curr != None:
    array.append(curr.val)
    curr = curr.next
  return array

File: src/languagesMap/js/ListNode/implementation.js
class ListNode {
  constructor(val, next) {
    this.val = (val === undefined ? 0 : val)
    this.next = (next === undefined ? null : next)
  }
}

function arrayToLinkedList(arrayNode) {
  if (!arrayNode.length) return null;
  const head = new ListNode()
  let next = head
  arrayNode.forEach((node, indice) => {
    next.val = node
    if (indice === arrayNode.length - 1) {
      next.next = null
    } else {
      next.next = new ListNode()
      next = next.next
    }
  })
  return head
}

function linkedListToArray(head) {
  const array = []
  let curr = head
  while (curr) {
    array.push(curr.val)
    curr = curr.next
  }
  return array
}
